sdk/servicebus/service-bus/src/core/linkEntity.ts

```diff
@@ -262,13 +262,59 @@ export abstract class LinkEntity<LinkT extends Receiver | AwaitableSender | Requ
 
       this._logger(`${this._logPrefix} Link has been created.`);
     } catch (err) {
-      await this.closeLink("linkonly");
+      await this.closeLink();
       throw err;
     } finally {
       this._isConnecting = false;
     }
   }
 
+  /**
+   * Clears token remewal for current link, removes current LinkEntity instance from cache,
+   * and closes the underlying AMQP link.
+   * Once closed, this instance of LinkEntity is not meant to be re-used.
+   */
+  async close(): Promise<void> {
+    // Set the flag to indicate that this instance of LinkEntity is not meant to be re-used.
+    this._wasClosedPermanently = true;
+
+    log.error(
```

Comment:
Should probably use this._logger so it goes to the right entity logging stream.

Reply:
There's another log below as well that we probably want to do the same thing for.

Reply:
(also, might as well swap over to the log prefix logger for the message)

Reply:
Not sure if we have logged an issue for this yet, but for Track 2 we will be moving away from the different logging streams for sender/receiver etc to logging stream based on log levels like info/error/warning. In that world, this would fall under "verbose", so I am using the `error` stream for now instead of the individual sender/receiver logging strea,

Reply:
Makes sense.